core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -0,0 +1,120 @@
+            $scope, AudioTranslationManagerService, AudioPlayerService) {
+          var filenameOfLastStartedAudio;
+
+          $scope.AUDIO_SETTINGS_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/settings.svg'));
+
+          var PLAY_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/speaker-not-playing.svg'));
+
+          var PAUSE_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/speaker-playing.svg'));
+
+          $scope.REWIND_AUDIO_BUTTON_IMAGE_URL = (
+            UrlInterpolationService.getStaticImageUrl(
+              '/icons/rewind-five.svg'));
+
+          $scope.playPauseAudioTranslation = function() {
+            // TODO(tjiang11): Change from on-demand loading to pre-loading.
+
+            // TODO(tjiang11): On first play, ask learner to pick language
+            // and subsequently for confirmation to use bandwidth 
+            // to download audio files.
+
+            $scope.extraAudioControlsAreShown = true;
+
+            if (!AudioPlayerService.isPlaying()) {
+              if (AudioPlayerService.isTrackLoaded() &&
+                  isRequestForSameAudioAsLastTime()) {
+                AudioPlayerService.play();
+              } else {
+                loadAndPlayAudioTranslation();
+              }
+            } else {
+              AudioPlayerService.pause();
+              if (!isRequestForSameAudioAsLastTime()) {
+                loadAndPlayAudioTranslation();
+              }
+            }
+          };
+
+          var isRequestForSameAudioAsLastTime = function() {
+            var requestedAudioTranslation =
+              $scope.audioTranslations[
+                AudioTranslationManagerService.getCurrentAudioLanguageCode()];
+            return filenameOfLastStartedAudio ==
+              requestedAudioTranslation.filename;
+          };
+
+          var loadAndPlayAudioTranslation = function() {
+            var currentAudioLanguageCode =
+              AudioTranslationManagerService.getCurrentAudioLanguageCode();
+
+            // TODO(tjiang11): If audio translation is not available
+            // in the current language, then inform the learner with
+            // a piece of text below the audio controls.
+            var audioTranslation =
+              $scope.audioTranslations[currentAudioLanguageCode];
+
+            if (audioTranslation) {
+              filenameOfLastStartedAudio = audioTranslation.filename;
+              AudioPlayerService.load(
+                audioTranslation.filename).then(function() {
+                  AudioPlayerService.play();
+                });
+            }
+          };
+
+          $scope.rewindAudioFiveSec = function() {
+            AudioPlayerService.rewind(5);
+          };
+
+          $scope.getPlayPauseButtonImageUrl = function() {
+            return AudioPlayerService.isPlaying() && 
```

Comment:
DIscussed with Tony -- each directive generates an ID on initialization, and when it wants to play some audio it passes that ID into AudioPlayerService. Directives can then check the "currently playing ID" in AudioPlayerServices to see if they should be displaying the play or pause icon.